crates/provider/src/native.rs

```diff
@@ -296,19 +287,85 @@ impl<T: FileSystem + Send + Sync> Provider for NativeProvider<T> {
         Ok(format!("tail -f {}/{}.log", self.tmp_dir, name))
     }
 
+    // TODO: Add test
     async fn pause(&self, node_name: &str) -> Result<(), ProviderError> {
-        // TODO: impl using run_command
+        let process_id: Result<u32, ProviderError> = match self.process_map.get(node_name) {
+            Some(process) => Ok(process.pid),
+            None => Err(ProviderError::MissingNodeInfo(
+                node_name.to_owned(),
+                "process".into(),
+            )),
+        };
+
+        let _ = self
+            .run_command(
+                vec![format!("kill -TSTP {}", process_id.unwrap())],
+                NativeRunCommandOptions {
+                    is_failure_allowed: true,
+                },
+            )
+            .await?;
         Ok(())
     }
 
+    // TODO: Add test
     async fn resume(&self, node_name: &str) -> Result<(), ProviderError> {
-        // TODO: impl using run_command
+        let process_id: Result<u32, ProviderError> = match self.process_map.get(node_name) {
+            Some(process) => Ok(process.pid),
+            None => Err(ProviderError::MissingNodeInfo(
+                node_name.to_owned(),
+                "process".into(),
+            )),
+        };
+
+        let _ = self
+            .run_command(
+                vec![format!("kill -CONT {}", process_id.unwrap())],
```

Comment:
`unwrap` will panic here if the `node_name` is not in the process_map. You can use `ok_or` and bubble up the error like this

```rust
        let process = self.process_map.get(node_name).ok_or(ProviderError::MissingNodeInfo(
            node_name.to_owned(),
            "process".into(),
        ))?;

        let _ = self
            .run_command(
                vec![format!("kill -CONT {}", process.pid)],
                NativeRunCommandOptions {
                    is_failure_allowed: true,
                },
            )
            .await?;
        Ok(())
  ```